src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -525,6 +598,73 @@ public bool TryGetRestClients(string requestPath, [MaybeNullWhen(false)] out Has
+        private string? GetDefaultNameFromConfiguration(OperationSet operationSet, ResourceTypeSegment resourceType)
+        {
+            if (_context.Configuration.MgmtConfiguration.RequestPathToResourceName.TryGetValue(operationSet.RequestPath, out var name))
+                return name;
+            if (_context.Configuration.MgmtConfiguration.RequestPathToResourceName.TryGetValue($"{operationSet.RequestPath}|{resourceType}", out name))
+                return name;
+
+            return null;
+        }
+
+        private string EnsureResourceDefaultName(string candidateName, OperationSet operationSet, RequestPath requestPath)
+        {
+            // read configuration to see if we could get a configuration for this resource
+            var resourceType = requestPath.GetResourceType(_mgmtConfiguration);
+            var defaultNameFromConfig = GetDefaultNameFromConfiguration(operationSet, resourceType);
+            if (defaultNameFromConfig != null)
+                return defaultNameFromConfig;
+
+            var resourcesWithSameName = _resourceDataSchemaNameToOperationSets[candidateName];
+            var resourcesWithSameType = OperationSets.SelectMany(opSet =>
+            {
+                var operation = opSet.Operations.First();
+                return GetRequestPath(opSet[operation], operation).Expand(_mgmtConfiguration);
+            }).Where(rqPath =>
+            {
+                return rqPath.Count % 2 == 0 && rqPath.GetResourceType(_mgmtConfiguration).Equals(resourceType);
+            });
+
+            var isById = operationSet.IsById(_context);
+            int countOfSameResourceDataName = resourcesWithSameName.Count();
+            int countOfSameResourceTypeName = resourcesWithSameType.Count();
+            if (!isById)
+            {
+                // this is a regular resource and the name is unique
+                if (countOfSameResourceDataName == 1)
+                    return candidateName;
+
+                // if countOfSameResourceDataName > 1, we need to have the resource types as the resource type name
+                // if we have the unique resource type, we just use the resource type to construct our resource type name
+                var types = resourceType.Types;
+                var name = string.Join("", types.Select(segment => segment.ConstantValue.LastWordToSingular().FirstCharToUpperCase()));
+                if (countOfSameResourceTypeName == 1)
+                    return name;
+
+                var parentOperationSet = requestPath.GetScopePath().GetResourceType(_mgmtConfiguration).Last().ConstantValue;
+                // if countOfSameResourceTypeName > 1, we will have to add the scope as prefix to fully qualify the resource type name
+                // first we try to add the parent name as prefix
+                if (!resourcesWithSameType.All(rqPath => rqPath.Equals(requestPath)))
+                    return $"{parentOperationSet.FirstCharToUpperCase().ToSingular()}{name}";
```

Comment:
We should use `LastWordToSingular`
```suggestion
                    return $"{parentOperationSet.LastWordToSingular().FirstCharToUpperCase()}{name}";
```